packages/block-library/src/post-author/edit.js

```diff
@@ -1,25 +1,294 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
-import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import apiFetch from '@wordpress/api-fetch';
+import { useRef } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
```

Comment:
> Is this a custom hook?

Yes, we have a few of those (see also: `__experimentalUseGradient`, `__experimentalUseEntitySaving`).

> How did we get around the hook rule 

This is a convention*, rather than one of the [essential rules](https://reactjs.org/docs/hooks-rules.html), and isn't enforced. (And I'm not sure how React could practically enforce it at runtime.)

*: And a good convention, but our `__experimental` prefixing takes precedence.